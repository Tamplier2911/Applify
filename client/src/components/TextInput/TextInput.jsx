// import './TextInput.scss'
import React from "react";
import PropTypes from "prop-types";

// JS Rendering CSS
import {
  TextInputContainer,
  TextInputArea,
  TextInputAreaLabel,
  FormInputFillingBar,
} from "./TextInputStyles";

const TextInput = ({ onInputChange, value, max, label, ...otherProps }) => {
  return (
    <TextInputContainer data-test="text-input">
      <TextInputArea
        onChange={onInputChange}
        {...otherProps}
        // defaultValue={value}
        value={value}
        maxLength={max}
      />
      <FormInputFillingBar />
      {label ? (
        <TextInputAreaLabel data-test="text-input-label" inputlength={value}>
          {label}
        </TextInputAreaLabel>
      ) : null}
    </TextInputContainer>
  );
};

TextInput.propTypes = {
  onInputChange: PropTypes.func,
  value: PropTypes.string,
  max: PropTypes.string,
  label: PropTypes.string,
  otherProps: PropTypes.object,
};

export default TextInput;
