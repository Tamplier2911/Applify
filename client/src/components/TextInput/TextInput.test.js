import React from "react";
import toJson from "enzyme-to-json";

// component
import TextInput from "./TextInput";

// utils
import { setShallow, findByTestAttr, checkProps } from "../../tests/testUtils";

describe("<TextInput />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallow(<TextInput />);
  });

  it("renders without an error", () => {
    const textInput = findByTestAttr(wrapper, "text-input");
    expect(textInput).toHaveLength(1);
  });

  it("renders children - label if label prop is passed", () => {
    wrapper = setShallow(<TextInput label="test label" />);
    const textInputLabel = findByTestAttr(wrapper, "text-input-label");
    expect(textInputLabel).toHaveLength(1);
  });

  it("recieves expected props", () => {
    const expectedProps = {
      onInputChange: () => {},
      value: "",
      max: "2",
      label: "",
      otherProps: {},
    };
    const err = checkProps(TextInput, expectedProps, "props");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const textInput = findByTestAttr(wrapper, "text-input");
    expect(toJson(textInput)).toMatchSnapshot();
  });
});
