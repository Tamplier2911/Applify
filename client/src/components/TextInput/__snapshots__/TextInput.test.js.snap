// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> matches snapshot 1`] = `
<styled.div
  data-test="text-input"
>
  <styled.textarea />
  <styled.div />
</styled.div>
`;
